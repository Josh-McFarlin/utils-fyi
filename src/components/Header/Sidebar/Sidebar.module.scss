@import "../../../../public/static/styles/Colors";

.hidden {
  display: none !important;
}

.root {
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 10;
  display: flex;
  flex-direction: row;
  background: rgba(21, 21, 21, 0.5);
  cursor: pointer;

  .sidebar {
    width: 450px;
    max-width: 80%;
    height: 100%;
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    padding: 8px 16px;
    margin-bottom: 16px;
    background: $secondary-bg-color;
    overflow-y: auto;
    cursor: initial;

    .logo {
      text-align: center;
      margin-bottom: 16px;
    }

    .route {
      padding: 12px 0;
      text-align: center;
      border: 2px solid white;
      border-radius: 4px;
      text-decoration: none;
      cursor: pointer;
      margin-bottom: 12px;
      background: none;
    }
  }
}
