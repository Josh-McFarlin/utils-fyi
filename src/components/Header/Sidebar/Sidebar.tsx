import * as React from "react";
import clsx from "clsx";
import Link from "next/link";
import routes from "../../../utils/routes";
import classes from "./Sidebar.module.scss";

interface Props {
  open: boolean;
  handleClose: () => void;
}

const Sidebar = ({ open, handleClose }: Props) => (
  <div
    className={clsx(classes.root, !open && classes.hidden)}
    role="button"
    tabIndex={0}
    onClick={handleClose}
    onKeyPress={(event) => event.stopPropagation()}
  >
    <div
      className={classes.sidebar}
      role="button"
      tabIndex={0}
      onClick={(event) => event.stopPropagation()}
      onKeyPress={(event) => event.stopPropagation()}
    >
      <h1 className={classes.logo}>utils-fyi</h1>
      {routes.map((route) => (
        <Link key={route.link} href={route.link} prefetch={false}>
          <button className={classes.route} onClick={handleClose} tabIndex={0}>
            {route.title}
          </button>
        </Link>
      ))}
    </div>
  </div>
);

export default Sidebar;
