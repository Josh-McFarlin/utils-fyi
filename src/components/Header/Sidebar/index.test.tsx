import React from "react";
import { render } from "../../../../test/testUtils";
import Sidebar from "./Sidebar";

// eslint-disable-next-line @typescript-eslint/no-empty-function
const noOp = () => {};

describe("Sidebar Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Sidebar open handleClose={noOp} />, {});

    expect(asFragment()).toMatchSnapshot();
  });
});
