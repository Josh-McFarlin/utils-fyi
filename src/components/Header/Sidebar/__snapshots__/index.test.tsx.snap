// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
    role="button"
    tabindex="0"
  >
    <div
      class="sidebar"
      role="button"
      tabindex="0"
    >
      <h1
        class="logo"
      >
        utils-fyi
      </h1>
      <button
        class="route"
        tabindex="0"
      >
        Home
      </button>
      <button
        class="route"
        tabindex="0"
      >
        UUID Generator
      </button>
    </div>
  </div>
</DocumentFragment>
`;
